src/devices/cpu/z80/z80.cpp

```diff
@@ -910,101 +803,84 @@ inline void z80_device::neg()
  ***************************************************************/
 inline void z80_device::daa()
 {
-	uint8_t a = A;
+	u8 a = A;
 	if (F & NF)
 	{
-		if ((F&HF) | ((A&0xf)>9)) a-=6;
-		if ((F&CF) | (A>0x99)) a-=0x60;
+		if ((F & HF) | ((A & 0xf) > 9))
+			a -= 6;
+		if ((F & CF) | (A > 0x99))
+			a -= 0x60;
 	}
 	else
 	{
-		if ((F&HF) | ((A&0xf)>9)) a+=6;
-		if ((F&CF) | (A>0x99)) a+=0x60;
+		if ((F & HF) | ((A & 0xf) > 9))
+			a += 6;
+		if ((F & CF) | (A > 0x99))
+			a += 0x60;
 	}
 
-	F = (F&(CF|NF)) | (A>0x99) | ((A^a)&HF) | SZP[a];
+	set_f((F & (CF | NF)) | (A > 0x99) | ((A ^ a) & HF) | SZP[a]);
 	A = a;
 }
 
 /***************************************************************
  * AND  n
  ***************************************************************/
-inline void z80_device::and_a(uint8_t value)
+inline void z80_device::and_a(u8 value)
 {
 	A &= value;
-	F = SZP[A] | HF;
+	set_f(SZP[A] | HF);
 }
 
 /***************************************************************
  * OR   n
  ***************************************************************/
-inline void z80_device::or_a(uint8_t value)
+inline void z80_device::or_a(u8 value)
 {
 	A |= value;
-	F = SZP[A];
+	set_f(SZP[A]);
 }
 
 /***************************************************************
  * XOR  n
  ***************************************************************/
-inline void z80_device::xor_a(uint8_t value)
+inline void z80_device::xor_a(u8 value)
 {
 	A ^= value;
-	F = SZP[A];
+	set_f(SZP[A]);
 }
 
 /***************************************************************
  * CP   n
  ***************************************************************/
-inline void z80_device::cp(uint8_t value)
+inline void z80_device::cp(u8 value)
 {
 	unsigned val = value;
-	uint32_t ah = AFD & 0xff00;
-	uint32_t res = (uint8_t)((ah >> 8) - val);
-	F = (SZHVC_sub[ah | res] & ~(YF | XF)) |
-		(val & (YF | XF));
-}
-
-/***************************************************************
- * EX   AF,AF'
- ***************************************************************/
-inline void z80_device::ex_af()
-{
-	PAIR tmp;
-	tmp = m_af; m_af = m_af2; m_af2 = tmp;
-}
-
-/***************************************************************
- * EX   DE,HL
- ***************************************************************/
-inline void z80_device::ex_de_hl()
-{
-	PAIR tmp;
-	tmp = m_de; m_de = m_hl; m_hl = tmp;
+	u32 ah = AFD & 0xff00;
+	u32 res = (u8)((ah >> 8) - val);
+	set_f((SZHVC_sub[ah | res] & ~(YF | XF)) | (val & (YF | XF)));
 }
 
 /***************************************************************
  * EXX
  ***************************************************************/
 inline void z80_device::exx()
 {
-	PAIR tmp;
-	tmp = m_bc; m_bc = m_bc2; m_bc2 = tmp;
-	tmp = m_de; m_de = m_de2; m_de2 = tmp;
-	tmp = m_hl; m_hl = m_hl2; m_hl2 = tmp;
+	std::swap(m_bc, m_bc2);
+	std::swap(m_de, m_de2);
+	std::swap(m_hl, m_hl2);
```

Comment:
The standard C++ idiom for swappable types is to do this as:
```c++
	using std::swap;
	swap(m_bc, m_bc2);
	swap(m_de, m_de2);
	swap(m_hl, m_hl2);
```

The reason being that if you do it this way, then if a specialisation for `swap` is defined in the namespace containing `PAIR`, dependent name lookup will prefer it over the blind implementation of `std::swap` using a temporary.  (I realise it won’t really make a difference in practice for this particular case.)